tests/runtime-trace-tests/tracerunner.ts

```diff
@@ -0,0 +1,309 @@
+/// <reference path="../../built/pxtcompiler.d.ts"/>
+/// <reference path="../../built/pxtsim.d.ts"/>
+
+import * as fs from 'fs';
+import * as path from 'path';
+import { exec } from 'child_process';
+
+import "mocha";
+import * as chai from "chai";
+
+import { TestHost } from "../common/testHost";
+import * as util from "../common/testUtils";
+import { resolve } from 'url';
+import { promisify } from 'util';
+
+// TODO: Split this file up. I (dazuniga) spent ~1hr trying to split this file into two pieces but I cannot figure out 
+// how to make that work with the way we're using jake & namespaces / modules
+
+// setup
+function initGlobals() {
+    Promise = require("bluebird");
+    let g = global as any
+    g.pxt = pxt;
+    g.ts = ts;
+    g.pxtc = pxtc;
+    g.btoa = (str: string) => new Buffer(str, "binary").toString("base64");
+    g.atob = (str: string) => new Buffer(str, "base64").toString("binary");
+}
+initGlobals();
+
+// Just needs to exist
+pxt.setAppTarget(util.testAppTarget);
+
+// tests
+const casesDir = path.join(process.cwd(), "tests", "runtime-trace-tests", "cases");
+
+describe("convert between ts<->py ", () => {
+    // TODO: can this be moved to a mocha before() block?
+    let tsAndPyFiles: string[]
+
+    pxsim.initCurrentRuntime = pxsim.initBareRuntime
+    cleanup()
+
+    const isDisabled = (file: string): boolean =>
+        path.basename(file).indexOf("TODO") >= 0
+
+    let tsFiles = util.getFilesByExt(casesDir, ".ts")
+    let pyFiles = util.getFilesByExt(casesDir, ".py")
+    tsAndPyFiles = tsFiles.concat(pyFiles)
+        .filter(f => !isDisabled(f))
+
+    tsAndPyFiles.forEach(file => {
+        it("should preserve the runtime semantics of " + path.basename(file), async function () {
```

Comment:
@riknoll here is where the actual mocha cases are generated. See my other comments as to what's going on